### ISSUE TITLE
bug report (verified): $downloadsLeft does not check for "Unlimited" downloads option

### ISSUE THREAD

**Issue by rumbdev** [scandipwa/customer-downloadable-graphql#4]
**Prerequisites:**   
 source: vendor/scandipwa/customer-downloadable-graphql/src/etc/di.xml
```
 <preference for="Magento\Downloadable\Controller\Download\Link"
                type="ScandiPWA\CustomerDownloadableGraphQl\Controller\Download\Link"/>
```

Class \ScandiPWA\CustomerDownloadableGraphQl\Controller\Download\Link line 93:
```
$downloadsLeft = $linkPurchasedItem->getNumberOfDownloadsBought() -
            $linkPurchasedItem->getNumberOfDownloadsUsed();
```

this checks only for "number" of downloads but does not check if "Unlimited" is selected in product's "downloadable information" section.
If we rely on "unlimited" it fails as when link expires.
See attachment: how is it possible to have "expired" if "Remaining Downloads" are set to "Unlimited"?

This happens because CLEARLY the scandi class takes another approach compared to the magento class it rewrites:
The magento class Magento\Downloadable\Controller\Download\Link on line 100 checks:
```
 if ($status == PurchasedLink::LINK_STATUS_AVAILABLE && ($downloadsLeft ||
            $linkPurchasedItem->getNumberOfDownloadsBought() == 0)
        ) 
```

ScandiPWA\CustomerDownloadableGraphQl\Controller\Download\Link checks instead
```
 if (($downloadsLeft || $linkPurchasedItem->getNumberOfDownloadsBought() === 0)) 
```

there is a HUGE difference here between == and === 0
basically it boils down to this.

because the value here (for various reason) could be a string if not a number in the original array (remember php would not throw an error when adding a string that can be converted to a number and would take care of it)

thanks

For the same reason see here how link is marked as expired even if "remaining downloads" shows "Unlimited left"
![unlimited-expired](https://user-images.githubusercontent.com/11228308/143463890-95152b97-3710-4da1-ad3b-42d52bb65aa7.png)

**Comment by rumbdev** [comment]
I tested the hotfix and created a pull request
